src/Float.mo

```diff
@@ -606,7 +664,24 @@ module {
   /// Float.compare(0.123, 0.1234) // => #less
   /// ```
   public func compare(x : Float, y : Float) : { #less; #equal; #greater } {
-    if (x < y) { #less } else if (x == y) { #equal } else { #greater }
+    let isNegative = func(number : Float) : Bool {
+      copySign(1.0, number) < 0.0
+    };
```

Comment:
Better to hoist this out as a private function or it might get allocated on each call to compare (unless the compiler is smart enough to optimize that).

Reply:
Thanks for this suggestion on performance improvement. I adjusted it.